app/controllers/reviews_controller.rb

```diff
@@ -13,6 +14,26 @@ def create
     end
   end
 
+  def edit
+    @review = Review.find(params[:id])
+    @park = @review.park
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    @park = @review.park
+
+    @review.update(review_params)
+
+    if @review.save
+      flash[:notice] = "You have successfully updated your review!"
+      redirect_to @park
+    else
+      flash.now[:notice] = "Invalid input."
+      render 'reviews/edit'
+    end
+  end
+
```

Comment:
Notice that `@review = Review.find(params[:id])` and `@park = @review.park` are happening multiple times? DRYing up controllers can help encapsulate these references to other classes:

```
class ReviewsController
  def edit
    review
    park
  end

  def update
    review
    park
    review.update(review_params)

    if review.save
      ..
    else
      ...
    end
  end

  private

  def review
    @review ||= Review.find(params[:id])
  end

  def park
    @park ||= review.park
  end
end
```